jax_md/energy.py

```diff
@@ -47,6 +47,27 @@ def _canonicalize_displacement_or_metric(displacement_or_metric):
     'than 4.')
 
 
+def spring(dr, length=f32(1), epsilon=f32(1), alpha=f32(2)):
```

Comment:
I'm not wild about the name of this function. The word "spring" has a physical meaning that, in principle, is much more general than this specific class of potentials. For example, you can imagine a bistable spring or an asymmetric spring. While you could probably make this same argument for the name "soft_sphere", I think it's ok there because there is a large literature base that refers to that class of potentials as "soft spheres."